clang/test/CIR/CodeGen/unary.cpp

```diff
@@ -155,28 +155,28 @@ int *inc_p(int *i) {
 
 void floats(float f) {
 // CHECK: cir.func @{{.+}}floats{{.+}}
-  +f; // CHECK: %{{[0-9]+}} = cir.unary(plus, %{{[0-9]+}}) : !cir.float, !cir.float
-  -f; // CHECK: %{{[0-9]+}} = cir.unary(minus, %{{[0-9]+}}) : !cir.float, !cir.float
+  f = +f; // CHECK: %{{[0-9]+}} = cir.unary(plus, %{{[0-9]+}}) : !cir.float, !cir.float
```

Comment:
Do you need to change the source here because otherwise these operations are removed? Does everything else in the function touching `f` gets removed or just this one? Perhaps the function should return a float instead so that the value is used? If simplification is getting too aggresive, perhaps we should instead tie it up with `O1` or higher, because `-O0` shouldn't delete as much operations. Example: https://godbolt.org/z/6zP7qnjdP